Proteomics/ProteomicsExtenstionMethods.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Proteomics
+{
+    public static class ProteomicsExtenstionMethods
+    {
+        public static List<int> AllIndexesOf(this string str, string value)
+        {
+            if (String.IsNullOrEmpty(value))
+                throw new ArgumentException("the string to find may not be empty", "value");
+            List<int> indexes = new List<int>();
+            for (int index = 0; ; index += value.Length)
+            {
+                index = str.IndexOf(value, index);
+                if (index == -1)
+                    return indexes;
+                indexes.Add(index);
+            }
+        }
+
+        public static void Fill<T>(this T[] originalArray, T with)
+        {
+            for (int i = 0; i < originalArray.Length; i++)
+            {
+                originalArray[i] = with;
+            }
+        }
+
+        public static string Reverse(string s)
```

Comment:
Not sure this needs a method since it can be done in a single line:
` string forward = "forward";
            string reverse = new string(forward.Reverse().ToArray());`